base/bootstrap.go

```diff
@@ -227,11 +227,16 @@ func (cc *CouchbaseCluster) getBucket(bucketName string) (b *gocb.Bucket, teardo
 	b = connection.Bucket(bucketName)
 	err = b.WaitUntilReady(time.Second*10, &gocb.WaitUntilReadyOptions{
 		DesiredState:  gocb.ClusterStateOnline,
-		RetryStrategy: gocb.NewBestEffortRetryStrategy(nil),
+		RetryStrategy: &goCBv2FailFastRetryStrategy{},
 		ServiceTypes:  []gocb.ServiceType{gocb.ServiceTypeKeyValue},
 	})
 	if err != nil {
 		_ = connection.Close(&gocb.ClusterCloseOptions{})
+
+		if errors.Is(err, gocb.ErrAuthenticationFailure) {
```

Comment:
Does gocb differentiate between authentication and authorization failure, or do we get ErrAuthenticationFailure in both cases?

Reply:
I don't think it does. Tried with with a bootstrap details that could only access a couple of buckets and got this failure.